src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENT_DATA = {};
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity >= 0) {
+            elementData = new Object[initialCapacity];
```

Comment:
How will you increase array size if it will be 0 or 1? Or how much time do you need to create new array if you start size will be 2? 
```suggestion
        if (initialCapacity >= DEFAULT_CAPACITY) {
            elementData = new Object[initialCapacity];
        } else if (initialCapacity > 0) {
           elementData = new Object()[DEFAULT_CAPACITY];
```